src/test/java/org/project/name/online/book/store/controller/BookControllerTest.java

```diff
@@ -0,0 +1,326 @@
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class BookControllerTest {
+    protected static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired DataSource dataSource,
+            @Autowired WebApplicationContext applicationContext
+    ) throws SQLException {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+        teardown(dataSource);
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/controller/"
+                            + "add-categories-to-categories-table.sql")
+            );
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/controller/"
+                            + "add-books-to-books-table.sql")
+            );
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/controller/"
+                            + "add-book_category-pairs-to-books_categories.sql")
+            );
+        }
+    }
+
+    @AfterAll
+    static void afterAll(@Autowired DataSource dataSource) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    private static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/controller/"
+                            + "remove-everything-after-tests.sql")
+            );
+        }
+    }
+
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    @Test
+    @DisplayName("Create a new book")
+    void createBook_ValidRequestDto_RespondStatusCreated() throws Exception {
+        CreateBookRequestDto requestDto = new CreateBookRequestDto();
+        requestDto.setTitle("Title");
+        requestDto.setAuthor("Author");
+        requestDto.setIsbn("123456789");
+        requestDto.setPrice(99.99);
+        requestDto.setDescription("Descr.");
+        requestDto.setCoverImage("Image");
+        requestDto.setCategoryIds(List.of(1L));
+        BookDto expected = new BookDto();
+        expected.setTitle(requestDto.getTitle());
+        expected.setAuthor(requestDto.getAuthor());
+        expected.setIsbn(requestDto.getIsbn());
+        expected.setPrice(requestDto.getPrice());
+        expected.setDescription(requestDto.getDescription());
+        expected.setCoverImage(requestDto.getCoverImage());
+        expected.setCategoryIds(List.of("Test name1"));
```

Comment:
it is difficult to read such tests
It is better to transfer all preparation to a private method with the appropriate name
fix it everywhere